source/core/ExecutionContext.cpp

```diff
@@ -329,6 +330,32 @@ InstructionCore* ExecutionContext::SuspendRunningQueueElt(InstructionCore* nextI
     }
 }
 
+void ExecutionContext::CopyAndEmptyRunningQueue(InstructionCore* nextInClump)
```

Comment:
The way pause is handled where we make a copy and empty the runQ is an internal implementation detail of the ExecutionContext. The public-facing API should just be called PauseExecution. Imagine tomorrow we change the implementation so that everything is handled with-in the runQ and we don't need a copy at all. That would force you to rename this API since it no longer matched the behavior.

Reply:
Same with the FillRunningQueue API. Externally it should just be a ResumeExecution API.